Fix broken implementation of irrKLang

gtryus opened the pull request:
make disposable
use IFileFactory to avoid locking file
Allow Thread.Sleep() logic to work.
Create test for longer sound playing.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/libpalaso/532)
<!-- Reviewable:end -->

papeh commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 121 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQHVDBp1YtWwZXTmzh:-KlQHVDBp1YtWwZXTmzi:b-6xfgu) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media/AudioIrrKlangSession.cs#L121)):*
> ```Smalltalk
> 			if (_sound != null)
> 			{
> 				_engine.RemoveAllSoundSources();
> ```

you `RemoveAllSoundSources` in a try-catch block below. Is that needed here?

---

*[SIL.Media/AudioIrrKlangSession.cs, line 152 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQH99sIUyvP5wllFcJ:-KlQH99sIUyvP5wllFcK:b-2ez5su) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media/AudioIrrKlangSession.cs#L152)):*
> ```Smalltalk
> 
> 			// myFile doesn't retain unicode characters so we capture it when the object is created
> 			public Stream openFile(string myFile)
> ```

if we don't use `myFile`, perhaps `dummy` would be a better name

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 49 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQHxydWIb7HPerWU5N:-KlQHxydWIb7HPerWU5O:bdk73n2) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L49)):*
> ```Smalltalk
> 			// ReSharper disable once UnusedVariable
> 			using (var x = AudioFactory.CreateAudioSession(path)) { }
> 			Assert.IsFalse(File.Exists(path));
> ```

does this `Assert` belong inside the `using` block?

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 277 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQLbxjPq9-P1olpOmz:-KlQLbxjPq9-P1olpOn-:bb4pyn3) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L277)):*
> ```Smalltalk
> 			//{
> 			//	_recorder.StopRecordingAndSaveAsWav();
> 			//}
> ```

commented code can be removed

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 288 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQLkmD1oN6wyY6OxD4:-KlQLkmD1oN6wyY6OxD5:b-ez6ftt) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L288)):*
> ```Smalltalk
> 					}
> 				}
> 				catch (Exception)
> ```

use `finally`

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 435 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQMvHrgNiMpOrNE503:-KlQMvHrgNiMpOrNE504:b9fedpx) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L435)):*
> ```Smalltalk
> 
> 		[Test]
> 		[Ignore("this test is to be run manually to test long recordings. Recite John 3:16")]
> ```

`[Test, Category("ByHand")]`?

I'm assuming the expected result is to hear both halves of the recording, punctuated by a half-second pause. This could use a brief instructional comment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/532)*
<!-- Sent from Reviewable.io -->

gtryus commented:
Review status: 4 of 7 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 121 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQHVDBp1YtWwZXTmzh:-KlUETbGBnkoyuR5VGYY:b-896fix) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media/AudioIrrKlangSession.cs#L121)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

you `RemoveAllSoundSources` in a try-catch block below. Is that needed here?
</blockquote></details>

Done.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 152 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQH99sIUyvP5wllFcJ:-KlUEVX7xpg2OHGwPZPl:b-896fix) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media/AudioIrrKlangSession.cs#L152)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

if we don't use `myFile`, perhaps `dummy` would be a better name
</blockquote></details>

Done.

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 49 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQHxydWIb7HPerWU5N:-KlUEYN266_dqzYJLVc9:b-1akoop) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L49)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

does this `Assert` belong inside the `using` block?
</blockquote></details>

I think this is right. I added a comment to clarify it.

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 277 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQLbxjPq9-P1olpOmz:-KlUF1RnMJGg0WJ70Xqg:b-896fix) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L277)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

commented code can be removed
</blockquote></details>

Done.

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 288 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQLkmD1oN6wyY6OxD4:-KlUF3-PCxd59fzv-055:b-896fix) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L288)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

use `finally`
</blockquote></details>

Done.

---

*[SIL.Media.Tests/AudioRecorderTests.cs, line 435 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/532#-KlQMvHrgNiMpOrNE503:-KlUF4jKWOqnua2bnauS:b-x5pzbc) ([raw file](https://github.com/sillsdev/libpalaso/blob/c85b74c891dd758d832a084f08f94ccb00a0eaad/SIL.Media.Tests/AudioRecorderTests.cs#L435)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

`[Test, Category("ByHand")]`?

I'm assuming the expected result is to hear both halves of the recording, punctuated by a half-second pause. This could use a brief instructional comment.
</blockquote></details>

I added the ByHand category but I don't want the test to run if the users haven't selected "exclude ByHand" so I left the Ignore. I added comments to explain that 10 seconds are recorded. Then 4 seconds are played followed by a pause followed by playing six seconds.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/532)*
<!-- Sent from Reviewable.io -->
